src/libcadet/model/binding/FreundlichLDFBinding.cpp

```diff
@@ -0,0 +1,205 @@
+				throw InvalidParameterException("FLDF_KKIN, FLDF_KF and FLDF_N,  have to have the same size");
+
+			return true;
+		}
+
+		inline const char* ExtFreundlichLDFParamHandler::identifier() CADET_NOEXCEPT { return "EXT_FREUNDLICH_LDF"; }
+
+		inline bool ExtFreundlichLDFParamHandler::validateConfig(unsigned int nComp, unsigned int const* nBoundStates)
+		{
+			if ((_kkin.size() != _kF.size()) || (_kkin.size() != _n.size()) || (_kkin.size() < nComp))
+				throw InvalidParameterException("FLDF_KKIN, FLDF_KF and FLDF_N,  have to have the same size");
+
+
+			return true;
+		}
+
+
+		/**
+		 * @brief Defines the multi component Freundlich LDF (Linear Driving Force) isotherm
+		 * @details Implements the Freundlich LDF adsorption model: \f[ \begin{align}
+		 *              \frac{\mathrm{d}q_i}{\mathrm{d}t} &= k_{f,i} c_{p,i}^{\frac{1}{n_{i}}} \end{align} \f]
+		 *         	Components without bound state (i.e., non-binding components) are supported.
+		 * @tparam ParamHandler_t Type that can add support for external function dependence
+		 */
+		template <class ParamHandler_t>
+		class FreundlichLDFBindingBase : public ParamHandlerBindingModelBase<ParamHandler_t>
+		{
+		public:
+
+			FreundlichLDFBindingBase() { }
+			virtual ~FreundlichLDFBindingBase() CADET_NOEXCEPT { }
+
+			static const char* identifier() { return ParamHandler_t::identifier(); }
+
+			virtual bool implementsAnalyticJacobian() const CADET_NOEXCEPT { return true; }
+
+			CADET_BINDINGMODELBASE_BOILERPLATE
+
+		protected:
+			using ParamHandlerBindingModelBase<ParamHandler_t>::_paramHandler;
+			using ParamHandlerBindingModelBase<ParamHandler_t>::_reactionQuasistationarity;
+			using ParamHandlerBindingModelBase<ParamHandler_t>::_nComp;
+			using ParamHandlerBindingModelBase<ParamHandler_t>::_nBoundStates;
+
+			template <typename StateType, typename CpStateType, typename ResidualType, typename ParamType>
+			int fluxImpl(double t, unsigned int secIdx, const ColumnPosition& colPos, StateType const* y,
+				CpStateType const* yCp, ResidualType* res, LinearBufferAllocator workSpace) const
+			{
+				typename ParamHandler_t::ParamsHandle const p = _paramHandler.update(t, secIdx, colPos, _nComp, _nBoundStates, workSpace);
+
+				/* Protein Flux: dq/dt = k_kin * (q* � q) with
+				q* = k_F * c^(1/n)
+				*/
+				unsigned int bndIdx = 0;
+				const double threshold = 1e-14;
+
+				for (int i = 0; i < _nComp; ++i)
+				{
+					// Skip components without bound states (bound state index bndIdx is not advanced)
+					if (_nBoundStates[i] == 0)
+						continue;
+					const double n_param = (double)static_cast<ParamType>(p->n[i]);
+					const double kF = (double)static_cast<ParamType>(p->kF[i]);
+					const double kkin = (double)static_cast<ParamType>(p->kkin[i]);
+					double const alpha_1 = ((2 * n_param - 1) / n_param) * kF * std::pow(threshold, (1 - n_param) / n_param);
+					double const alpha_2 = ((1 - n_param) / n_param) * kF * std::pow(threshold, (1 - 2 * n_param) / n_param);
```

Comment:
Please do not convert the `active` types to `double`! This will disable parameter sensitivities.
If you need to copy these to local variables, please do
```c++
const ParamType& n_param = static_cast<ParamType>(p->n[i]);
```

Reply:
Done